### ISSUE TITLE
ReactTestUtils.scryRenderedComponentsWithType having problem with compiled code?

### ISSUE THREAD

**Issue by samzhao** [facebook/react#4378]
EDIT: React version 0.13.3

Hi,

I was trying to use `scryRenderedComponentsWithType` in my test, but the result it returns isn't as expected.

Given that I have a `BaseComponent` that other components extends from, and I have

```javascript
class Project extends BaseComponent {
    render() {
        return (<h3>Projects</h3>)
    }
}
```

Now when I do

```javascript
let renderedEl = ReactTestUtils.renderIntoDocument(
    <div>
        <Project/>
    </div>
)

let projects = ReactTestUtils.scryRenderedComponentsWithType(renderedEl, Project)

console.log(projects)
```

I will get `[]`. The reason for that is this line https://github.com/facebook/react/blob/v0.13.3/src/test/ReactTestUtils.js#L85 evaluates to `false` because the `inst.constructor` is now `function b()` for some reason (I believe it's minified by Webpack), so of course it doesn't equal `function Project()`.

Is that something you guys are aware of, or did I use `scryRenderedComponentsWithType` incorrectly?

Thanks.

**Comment by zpao** [comment]
You probably just shouldn't test minified code. Our code there is depending on identity (at least for composite components) so it would have to be packaged together to properly work anyway. Even if they had the same name but were different copies of the same function, the test would fail.

I don't think there's anything for us to do here so closing out.

**Comment by samzhao** [comment]
@zpao ok that makes sense.

One interesting thing I just discovered when looking at the source here on github is that the `scryRenderedComponentsWithType` method in the master branch works. I took the content of the `isCompositeComponentWithType` method from master branch and use it in my project which uses React 0.13.3, and it works as well.

I can live with overriding the current `scryRenderedComponentsWithType` method as a workaround for now and remove it after 0.14.0 is released (or perhaps a 0.13.x patch). So just FYI.

Thanks for your quick response!
